org.eclipse.draw2d/src/org/eclipse/draw2d/EventListenerList.java

```diff
@@ -126,28 +83,14 @@ public synchronized <T> Iterable<T> getListenersIterable(final Class<T> listener
 	 * @param listener the listener
 	 */
 	public synchronized <T> void removeListener(Class<T> c, Object listener) {
-		if (array == null || array.length == 0) {
-			return;
-		}
 		if (listener == null || c == null) {
 			throw new IllegalArgumentException();
 		}
 
-		int index = 0;
-		while (index < array.length) {
-			if (array[index] == c && array[index + 1] == listener) {
-				break;
-			}
-			index += 2;
+		List<Object> specListeners = listeners.get(c);
+		if (specListeners != null) {
+			specListeners.remove(listener);
```

Comment:
Something else I just remembered when using classes as keys... Would it make sense to remove the key-value pair from the map when the number of listeners is empty, in order to remove any strong references to said classes?